CRUD.c

```diff
@@ -0,0 +1,186 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define FILENAME "users.txt"
+
+// User structure
+typedef struct {
+    int id;
+    char name[40];
+    int age;
+} User;
+
+// Let's check whether the file exists or not
+void checkFile() {
+    FILE *file = fopen(FILENAME, "a");
+    if (file == NULL) {
+        printf("Something's wrong.\n");
+        exit(1);
+    }
+    fclose(file);
+}
+
+// Create a new user
+void createUser() {
+    User user;
+    FILE *file = fopen(FILENAME, "a");
+    if (file == NULL) {
+        printf("Couldn't open the file. Try again later.\n");
+        return;
+    }
+
+    printf("Let's add a new user:\n");
+    printf("Enter User ID: ");
+    scanf("%d", &user.id);
+    printf("Enter name : ");
+    scanf(" %[^\n]s", user.name);
+    printf("Enter age (valid age please!): ");
+    scanf("%d", &user.age);
+
+    // Write user details to the file
+    fprintf(file, "%d,%s,%d\n", user.id, user.name, user.age);
+    fclose(file);
+
+    printf("User added successfully.\n");
+}
+
+// Display all users
+void displayUsers() {
+    FILE *file = fopen(FILENAME, "r");

```

Comment:
There is no need to open and close the file every time. This can be done only once.

Reply:
👍